src/test/ui/traits/issue-72410.rs

```diff
@@ -0,0 +1,11 @@
+// check-pass
+// Regression test for #72410, this should be used with debug assertion enabled.
+
+pub trait Foo {
+    fn map()
+    where
+        Self: Sized,
+        for<'a> &'a mut [u8]: ;
```

Comment:
We should have a test for a type that *is not* WF, which this PR would allow compiling, but we'd never want to.

Reply:
Should we separate that test in an another file? And I'm not sure what kind of test case is appropriate.

Reply:
Perhaps something like

```rust
trait Foo<'a> { }

fn map()
    where for<'a> &'a mut [dyn Foo<'a>]
```

which I think ought to fail because `dyn Foo<'a>: Sized` doesn't hold

Reply:
you could add it in the same file I suppose

Reply:
Thanks Niko! Added it.